promql/engine.go

```diff
@@ -2909,16 +2974,34 @@ func (ev *evaluator) aggregation(e *parser.AggregateExpr, q float64, inputMatrix
 				case h != nil:
 					// Ignore histograms for STDVAR and STDDEV.
 					group.seen = false
+					if op == parser.STDVAR {
+						annos.Add(annotations.NewHistogramIgnoredInAggregationInfo("stdvar", e.Expr.PositionRange()))
+					} else {
+						annos.Add(annotations.NewHistogramIgnoredInAggregationInfo("stddev", e.Expr.PositionRange()))
+					}
```

Comment:
Can we avoid the `if` by using `parser.ItemTypeStr[op]`?

Same for the `if`s below.

Reply:
`parser.ItemTypeStr` only contains the string representation of operators. There is an unexported list `key` in `parser` which is of type `map[string]int` so we can't use it like `parser.ItemTypeStr`. We can create another map for the aggregations but should we?

Reply:
Yeah, let's leave it alone for now.